ReactNativeClient/lib/joplin-renderer/MdToHtml/rules/copy_to_clipboard.js

```diff
@@ -0,0 +1,108 @@
+const fs = require('fs');
+const path = require('path');
+const filePath = path.join(__dirname, '../copyFunction.js');
+const btnCode = fs.readFileSync(filePath).toString();
+
+
+const copyCodeStyle = `
+.hljs {
+	position: relative;
+}
+.copy-wrapper {
+  display: none;
+  position: absolute;
+  top: 0;
+  right:0;
+}
+.copy-wrapper button {
+	padding: 8px 8px;
+	background: #fff;
+	margin: 5px 8px;
+	border: none;
+	cursor: pointer
+}
+.copy-wrapper button i {
+	font-size: 0.98rem;
+}
+
+  .hljs:hover .copy-wrapper{
+	  display: inline !important;
+  } 
+
+.u-mdic-copy-notify {
+	font-size: 0.75rem;
+	color: #cccccc;
+	display: none;
+	background: #fff;
+	border-radius: 10px;
+	padding: 5px 10px;
+}
+`;
+
+
+
+function generateUuid() {
+	return `${+Date.now()}-${parseInt(Math.random() * 100000)}`;
+}
+
+const LAST_TAG = '</pre>';
+
+const enhance = (render, options = {}) => (...args) => {
+	/* args = [tokens, idx, options, env, slf] */
+	const {
+		failText = 'copy fail', // copy-fail text
+		successText = 'copied!', // copy-success text
+		successTextDelay = 2000, // successText show time [ms]
+	} = options;
+	const [tokens = {}, idx = 0] = args;
```

Comment:
No default value please as the value should be there. If they aren't, you can throw an error (but probably not needed).

Reply:
defaults removed